README.md

```diff
@@ -1,17 +1,130 @@
+
+# 2 IPLD Schema
+
+Unlike a JWT, the IPLD encoding of UCAN does not require separate header, claims, and signature fields.
+
+```ipldsch
+type UCAN struct {
+  version String
+
+  issuer DID
+  audience DID
+  signature Signature
+
+  capabilities [Capability]
+  proofs [&UCAN]
+  expiration Int
+
+  facts [Fact]
+  nonce optional String
+  notBefore optional Int
+} representation map {
+  field facts default []
+  field proofs default []
+}
+```
+
+## 2.1 DID
+
+DIDs MUST be encoded as [DID](https://www.w3.org/TR/did-core/)s. The [`did:key` method](https://w3c-ccg.github.io/did-method-key/) is RECOMMENDED as it is self-contained.
+
+``` ipldsch
+type DID = String
+```
+
+## 2.2 Capability
+
+``` ipldsch
+type Capability struct {
+  with Resource
+  can Ability
+  extension optional Extension
+} representation map
+```
+
+### 2.2.1 Resource
+
+The resource pointer MUST be formatted as a [URI](https://www.rfc-editor.org/rfc/rfc3986).
+
+``` ipldsch
+type Resource = String
+```
+
+### 2.2.2 Ability
+
+Abilities MUST be lowercase, and MUST be namespaced with `/` delimeters. Abilities SHOULD have at least one path segment.
+
+``` ipldsch
+type Ability = String
+```
+
+### 2.2.3 Extensions
+
+The extended capability field is OPTIONAL. When present, it MUST contain any additional domain specific details and/or restrictions of the capability. 
+
+``` ipldsch
+type Extension = { String: Any }
+```
+
+## 2.3 Fact
+
+Facts MUST be structured as maps with string keys.
+
+``` ipldsch
+type Fact { String: Any }
+```
+
+## 2.4 Proof
+
+As of UCAN 0.9, all proofs MUST be given as links (CIDs). Note that UCANs MAY be inlined via the [identity multihash (`0x00`)](https://github.com/multiformats/multicodec/blob/master/table.csv#L2) CID.
+
+``` ipldsch
+&UCAN
+```
+
+## 2.5 Signature
+
+The signature MUST be computed by first encoding it as a [canonical JWT](#3-jwt-canonicalization), and then signed with the issuer's private key.
+
+``` ipldsch
+type Signature = Bytes
+```
+
+# 3 JWT Canonicalization
+
+Per the core UCAN spec, all implementations MUST support JWT encoding. This provides a common representation that all implementations can understand. JWT canonicalization allows for an IPLD UCAN to be expressed as a JWT, retain the JWT signature scheme, and so on for compatibility, while retaining the ability to translate into other formats for storage or transmission among IPLD-enabled peers.
+
+To canonicalize an IPLD UCAN to JWT, the JSON segments MUST be encoded per [`dag-json`](https://ipld.io/specs/codecs/dag-json/spec/), encoded as unpadded [base64url](https://datatracker.ietf.org/doc/html/rfc4648#section-5), and joined with `.`s.
```

Comment:
I'm afraid that is alone not enough, additional steps entail:

1. Using all lower case variant of `can` field.
2. Omit all optional fields if they're empty e.g. `fct`.